integration-testing/src/fixtures/common.py

```diff
@@ -57,33 +68,20 @@ def docker():
     docker_client.volumes.prune()
 
 
-@pytest.fixture(scope="package")
-def docker_network(docker):
-    network_name = f"rchain-{tools.random.random_string(5).lower()}"
-
-    docker.networks.create(network_name, driver="bridge")
-
-    yield network_name
-
-    for network in docker.networks.list():
-        if network_name == network.name:
-            logging.info(f"removing {network.name}")
-            network.remove()
-
 
 @contextmanager
-def create_bonds_file(validator_keys):
-    (fd, bonds_file) = tempfile.mkstemp(prefix="rchain-bonds-file-", suffix = ".txt", dir="/tmp")
+def bonds_file(validator_keys):
+    (fd, file) = tempfile.mkstemp(prefix="rchain-bonds-file-", suffix = ".txt", dir="/tmp")
     logging.info(f"Using bonds file: `{bonds_file}`")
 
     with os.fdopen(fd, "w") as f:
         for pair in validator_keys:
             bond = random.randint(1, 100)
             f.write(f"{pair.public_key} {bond}\n")
 
-    yield bonds_file
+    yield file
 
-    os.unlink(bonds_file)
+    os.unlink(file)
```

Comment:
I think the file won't be deleted if an exception gets thrown within the body of `with` statement that uses this context manager.  You may need a `try...finally` block around `yield`.

Reply:
The `@contextmanager` does that

Reply:
That's not what the example indicates: https://docs.python.org/3/library/contextlib.html#contextlib.contextmanager

There's a explicit try...finally there.

Reply:
oops! you are right. 

it seems I was using the contextmanager in a wrong way

Reply:
Just check yourself:

```
from contextlib import contextmanager

@contextmanager
def foo():
  print('prologue')
  yield 123
  print('epilogue')


with foo():
  raise ValueError("bar")
```

```
(dev)% python3 manager.py
prologue
Traceback (most recent call last):
  File "manager.py", line 11, in <module>
    raise ValueError("bar")
ValueError: bar
zsh: exit 1     python3 manager.py
```

Reply:
I tried that and you are right. Thanks